File: contrib/matlab/plot_spectrum.m
% Plots energy spectra as a function of time
% Type 'plot_spectrum' in MATLAB to execute

% Path to the data
path = '../../3D/bin/';

% Spatial resolution
N = 128;

% Array with wavenumbers starting at 1
k = [1:N/2+1];

% Reads and plots all spectra in the directory.
% We only plot one every five spectra, starting
% from the second.
filelist = dir(strcat(path,'kspectrum.*.txt'));
nfiles = numel(filelist);
figure(1)
for i = 2:5:nfiles,
  ene = load(strcat(path,filelist(i).name));
  loglog(k,ene)
  hold all
end
xlim([1 N/3])
xlabel('k')
ylabel('E(k)')
hold off

File: contrib/python/plot_spectrum.py
import numpy as np
import glob as glob
import matplotlib.pyplot as plt

# Plots energy spectra as a function of time
# Execute in ipython with '%run plot_spectrum.py'

# Path to the data
path = '../../3D/bin/'

# Reads and plots all spectra in the directory.
# We only plot one every five spectra, starting
# from the second.
filelist = sorted(glob.glob(path+'kspectrum.*.txt'))
nfiles = np.size(filelist)
plt.figure(1)
for i in range(1, nfiles, 5):
  ene = np.loadtxt(filelist[i])
  plt.loglog(ene[:,0],ene[:,1])
nmax = 2*(np.size(ene[:,0])-1)
plt.xlim(1,nmax/3)
plt.xlabel('k')
plt.ylabel('E(k)')
plt.show()
